runtime/fs_node.js

```diff
@@ -104,16 +121,161 @@ MlNodeDevice.prototype.open = function(name, f) {
     var fd = this.fs.openSync(this.nm(name), res);
     return new MlNodeFile(fd);
   } catch (err) {
-    caml_raise_sys_error(err.toString());
+    if (raise_unix) {
+      this.raise_unix_exn_of_nodejs_error(err);
+    } else {
+      caml_raise_sys_error(err.toString());
+    }
   }
 }
 
-MlNodeDevice.prototype.rename = function(o,n) {
+MlNodeDevice.prototype.rename = function(o, n, raise_unix) {
   try {
     this.fs.renameSync(this.nm(o), this.nm(n));
   } catch (err) {
-    caml_raise_sys_error(err.toString());
+    if (raise_unix) {
```

Comment:
I would refactor some of the error handling logic and lift things to a helper function.
```js
function raise_nodejs_error(err, raise_unix) {
    if (raise_unix) {
      this.raise_unix_exn_of_nodejs_error(err);
    } else {
      caml_raise_sys_error(err.toString());
    }
}
```

Reply:
Updated! I'm going to refactor this a bit more so the nodejs error transformation just happens in this helper.